ksql-engine/src/main/java/io/confluent/ksql/KsqlContext.java

```diff
@@ -75,20 +80,25 @@ public static KsqlContext create(
     final DefaultSchemaInjector schemaInjector = new DefaultSchemaInjector(
         new SchemaRegistryTopicSchemaSupplier(serviceContext.getSchemaRegistryClient()));
 
-    return new KsqlContext(serviceContext, ksqlConfig, engine, schemaInjector);
+    return new KsqlContext(
```

Comment:
The fact that we create the schema injector above, rather than passing in a factory, may be a bug.. (my fault probably).   Would you mind changing it to follow the same pattern as `StandaloneExecutor` and `KsqlResource`?